### DISCUSSION TITLE
🎉 New Release: Sharing, and Callout Import/Export (v1.14)

### DISCUSSION THREAD

**Discussion by chrisgurney** [chrisgurney/obsidian-note-toolbar#153] [Announcements]
Here's what's new in v1.14:

## Share toolbars

Share toolbars with a link that you can send to other users.

- Right-click on a toolbar in a note to see the _Share..._ option, or use the new "More" (...) menu for toolbars in Settings.
- When the recipient clicks on the link, they will be prompted for confirmation and see a preview of the toolbar.

→ Learn more about [how sharing works](https://github.com/chrisgurney/obsidian-note-toolbar/wiki/Sharing-toolbars).

<a href="https://github.com/user-attachments/assets/a1d59acb-729c-4017-83d9-99985a920cf9">
  <img src="https://github.com/user-attachments/assets/a1d59acb-729c-4017-83d9-99985a920cf9" width="700"/>
</a>

## Import toolbars and items from Note Toolbar Callouts

Create toolbars from [Note Toolbar Callouts](https://github.com/chrisgurney/obsidian-note-toolbar/wiki/Note-Toolbar-Callouts) in one of two ways:

1. import a callout into a new, or existing toolbar in Settings; or
2. select some Note Toolbar Callout markdown in a note, then right-click and _Create Note Toolbar from callout_.

<a href="https://github.com/user-attachments/assets/0b82104e-f5f4-40f9-9880-3e9425c5f908">
  <img src="https://github.com/user-attachments/assets/0b82104e-f5f4-40f9-9880-3e9425c5f908" width="700"/>
</a>

→ Learn more about [importing toolbars and items](https://github.com/chrisgurney/obsidian-note-toolbar/wiki/Creating-toolbars-from-callouts).

## Copy toolbars as callouts

Right-click on a toolbar to copy it as a Note Toolbar Callout, to your clipboard.

<a href="https://github.com/user-attachments/assets/f69ad45f-2b52-4c0d-a332-92c68e736c5d">
  <img src="https://github.com/user-attachments/assets/f69ad45f-2b52-4c0d-a332-92c68e736c5d" width="700"/>
</a>

→ Learn more about [copying toolbars as callouts](https://github.com/chrisgurney/obsidian-note-toolbar/wiki/Creating-callouts-from-toolbars), including what's supported and what's not.

## Commands to make building callouts easier

`Note Toolbar: Copy URI for command`

Lets you select a command and then copies its URI to the clipboard. Example output:

```markdown
obsidian://note-toolbar?command=workspace:toggle-pin
```

`Note Toolbar: Copy callout item for command`

Lets you select a command, and then copies a Note Toolbar Callout line to the clipboard. Example output:

```markdown
[]()<data data-ntb-command="workspace:toggle-pin"/> <!-- Toggle pin -->
```

...then all you need to do is choose what the item should look like in the link's text (`[]`).

## Updated toolbar examples

Check out the [expanded Examples page](https://github.com/chrisgurney/obsidian-note-toolbar/wiki/Examples) with numerous use cases and now easier-to-add toolbars. [Share your own examples!](https://github.com/chrisgurney/obsidian-note-toolbar/discussions/categories/show-and-tell).

## Other improvements

- In Settings, in the toolbar list, look for a new "More" (...) menu (where "Duplicate" used to be) containing _Duplicate_, _Share_, _Copy as callout_, and _Delete_ options.
- Most plugin startup tasks are now done after the layout's ready, and after initial toolbar rendering, which may help with startup performance. `#ObsidianOctober`
- An error message is now shown if a [Note Toolbar URI](https://github.com/chrisgurney/obsidian-note-toolbar/wiki/Note-Toolbar-URIs) contains invalid parameters.
- Quick Toolbars: Items are kept in order they're displayed in the toolbar, when browsing by toolbar. _Thanks to @Glint-Eye for the suggestion._
- Updated the _Help_ window with a link to a bug report Google form, removing the pre-requisite for a GitHub account.
- Updated the _What's New_ window with a permanent link to the Roadmap, at the bottom.

## Fixes

- Duplicated toolbar items without changes now update the settings UI, save the change, and re-render the toolbar.
- Toolbar icons/text weren't being vertically centered by a few pixels, due to a line-height being added.

## And in case you missed it...

- [v1.13 release](https://github.com/chrisgurney/obsidian-note-toolbar/releases/tag/1.13): _What's New_ + _Help_ windows, and access to toolbars for linked notes from the file menu.
- [v1.12 release](https://github.com/chrisgurney/obsidian-note-toolbar/releases/tag/1.12.1): Translations, and Quick Tools (fast access to all of your tools).

**Comment by chrisgurney** [comment]
@Moyf @laktiv @FelipeRearden

**Comment by Moyf** [comment]
Awesome!!

**Comment by chrisgurney** [comment]
v1.14.0 is now available via Community Plugins.

**Comment by laktiv** [comment]
Hi gents,
Would you be so kind to help me?
Can't import panel via url from [Examples](https://github.com/chrisgurney/obsidian-note-toolbar/wiki/Examples). But I know it works (you woudn't post it). Tried to open with by other vaults and one of old has imported. Two current has refused. Looks like this.
![image](https://github.com/user-attachments/assets/a5430943-dda9-4d39-8499-cd2847a7ce88)
Could you please tell me how to use proper the devtool? Still can't get it how to use it in a proper way. Mb you have a good like link to share? But for a dummy. )

Here are the screen and log. 
[obsidian.md-1730850516008.log](https://github.com/user-attachments/files/17642668/obsidian.md-1730850516008.log)
![image](https://github.com/user-attachments/assets/5bad66af-8032-405a-aeed-a8d6ebf8c0b4)
All links refers to some code in scripts which gives me no info. Why no other records like in "Network" tab? I was interested in API, there were info about data exchange. And, how do you read the successful operation?
![image](https://github.com/user-attachments/assets/4edf52fc-695d-4994-9717-789b6936abe2).

Hope you would be able to guide me in the right way. Because every step is a little pain.. You must to be a programmer to use Obsidian, but I want to insert my a bit custom templates as well. ) And almost all available manuals is not for the novice.

Thank you

**Reply by chrisgurney** [reply]
Hi @laktiv can you share the steps you took to get this error? What exactly did you click on? Is this using the beta?

**Reply by laktiv** [reply]
Sure.
1. This URL in Chrome - https://github.com/chrisgurney/obsidian-note-toolbar/wiki/Examples
2. Obsidian opened (offtop - tried few combinations, 1-2-3 opened instances at the same time, doesn't matter, opens (sends command) to last active. if not working, this is doesn't matter)
3. Tried to import panel by clicking on purple buttons below "Insert items", "Note Toolbar shortcuts", "Note tools", "Obsidian Help", "Search", "Switch themes and views". Only screenshot with blank window appeared.
4. Halt of panels have tried to import through callout text. No result, actually not anything happen at all - no reaction or feedback on pressing the button. it was strange behavior, because I definitely already used this functions.
5. Then I've opened 3rd old unused vault and it worked as supposed to with both methods.

Obsidian 1.7.5 (yesterday update)
Note Toolbar 1.15-beta-16 (both with custom updated translation file and after full reinstall)
Windows 11, tried reboot as well


Note Toolbar

**Reply by laktiv** [reply]
@chrisgurney update.
Delete - install non beta - working!
![image](https://github.com/user-attachments/assets/e10a8de2-e44d-4567-b1ed-1ca8f78f3e9f)

Brat have updated the plugin - stopped worked.
![image](https://github.com/user-attachments/assets/7f005914-7693-4932-a3e1-5a36e106f22d)

**Reply by laktiv** [reply]
Delete - install via Brat frozen 1.14.9 - back to work.
![image](https://github.com/user-attachments/assets/df218623-afb2-4c63-b874-0a4d07740859)

I thought it's on my side. But why 3rd vault have opened the link properly? Everything were with same updates..

**Reply by chrisgurney** [reply]
@laktiv This is fixed in the latest beta! Thank you for testing.

In hindsight this came from a global code change I made, without VS Code telling me there was a problem, and then I didn't regression test it. Lesson learned.

**Reply by chrisgurney** [reply]
To answer your question about debugging something like this in the future, my thoughts would be:
- Run the code for the plugin locally, so you can better see where in the code the problem is.
- Find YouTube videos about using DevTools. Really all you need to focus on is the debugging and CSS parts. 
- If you like, create a simple plugin. This is where I started: https://www.youtube.com/playlist?list=PLIDCb22ZUTBnMCbJa-st4PD5T3Olep078

**Reply by laktiv** [reply]
Hi @chrisgurney 
Thank you for your answer and details provided. As well as fix ;)
Great to have such quick reaction and fixing
